### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by someshsingh22**
1. Levenshtein Edit Distance Added with polynomial complexity (numpy optimized)
-	LCS and Sum normalization Available as arg

2. Jaccard Distance Done

3. Euclidean Distances complete
-	normalization available
-	CountVectorzier Space issue resolved (only numpy is used)

### CODE REVIEW

File: decepticonlp/metrics/char_metrics.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    #Normalize LCS - Max Strategy
+    print(levenshtein("HeyS", "HEY", normalize="lcs"))
+    0.75
+
+
+    :params
+    :text1, text2 - Both the inputs
+    :normalize - Pass "none" for getting the raw distance
+               - Pass "sum" for getting Normalized Levenshtein Distance
+               - Pass "lcs" for getting #Normalize LCS
```
Comment (Sharad24) [inline decepticonlp/metrics/char_metrics.py:27]: This is not the correct format I think.
It should be 
```
        :param a: First Letter
        :param B: Second Letter
        :type a: char
        :type B: char 
        :returns: z based on the Q table 
        :rtype: char ```

**Comment by Sharad24** [comment]
Resolves #15

File: decepticonlp/metrics/char_metrics.py (hunk 1 above)
Comment (someshsingh22) [reply decepticonlp/metrics/char_metrics.py:27]: Nice I will update the changes

**Comment by Sharad24** [comment]
Ok sounds good. Put tests in `tests` though

**Comment by Sharad24** [comment]
Would be better if we follow the same directory structure in tests. So yours will be in `tests/test_metrics/....py`

**Comment by someshsingh22** [comment]
I dont get you @rajaswa, Where have we done the unit testing before ? I have tested all functionalities up to metrics from the root dir. there is no .tests rn I guess

**Comment by Sharad24** [comment]
He meant `tests` I guess.

**Comment by someshsingh22** [comment]
Ok, I actually don't know how that works, any link ?

**Comment by Sharad24** [comment]
What?

**Comment by someshsingh22** [comment]
Update Metrics are also added, for floating-point errors absolute error is checked with a tolerance of 1e-5

File: requirements.txt [hunk 2]
```diff
@@ -7,8 +7,6 @@ tox==3.14.0
 coverage==4.5.4
 Sphinx==1.8.5
 twine==1.14.0
-
-setuptools
```
Comment (rajaswa) [inline requirements.txt:11]: We do need setuptools for local development
Add it back to requirements.txt

Comment (someshsingh22) [reply requirements.txt:11]: Damn, I didnt even see that

**Pull request merged**